src/web/js/check-ui.js

```diff
@@ -24,246 +24,426 @@
+            this.block.needRefreshing.push(snippet.editor);
           }
-        });
+        };
 
-        var testsInBlock = 0,
-        testsPassingInBlock = 0;
-
-        function makeTitle(loc, passed) {
-          var testName =
-            $("<a>")
-            .text("Test " + testsInBlock)
-            .addClass("hinted-highlight");
-
-          var header =
-            $("<header>")
-            .append(testName)
-            .append(": " + (passed ? "Passed" : "Failed"))
-            .attr('title',"Click to scroll editor to test.");
-
-          var source = get(get(checkBlock, "loc"), "source");
-          if(editors.hasOwnProperty(source)){
-            var handle =  editors[source].markText(
-                             { line: get(get(checkBlock, "loc"), "start-line") - 1,
-                               ch:   get(get(checkBlock, "loc"), "start-column") },
-                             { line: get(get(checkBlock, "loc"), "end-line") - 1,
-                               ch:   get(get(checkBlock, "loc"), "end-column") },
-                             { inclusiveLeft:false,
-                               inclusiveRight:false,
-                               type:"range" });
-            header.on("click", function(e){
-              var handleLoc = handle.find();
-              if(source === "definitions://" && handleLoc !== undefined) {
-                editors[source].scrollIntoView(handleLoc.from.line, 100);
-              } else if (source.indexOf("interactions") != -1) {
-                editors[source].getWrapperElement().scrollIntoView(true);
-              }
+        return PassingTestSkeleton;
+      }();
+
+      var expandedCheckBlock = undefined;
+
+      var CheckBlockSkeleton = function () {
+        function CheckBlockSkeleton(name, loc, tests, error) {
+          var _this = this;
+
+          var container = document.createElement("div");
+          var testList = document.createElement("div");
+          var testFrag = document.createDocumentFragment();
+          var header = document.createElement("header");
+          var summary = document.createElement("span");
+
+          for (var i = 0; i < tests.skeletons.length; i++) {
+            var test = tests.skeletons[i];
+            test.block = this;
+            testFrag.appendChild(test.container);
+          }
+
+          if (error !== undefined) {
+            summary.textContent = "An unexpected error halted the check-block before Pyret was finished with it. " + "Some tests may not have run.";
+            var errorTestsSummary = document.createTextNode("Before the unexpected error, " + tests.executed + (tests.executed === 0 ? " tests " : " test ") + "in this block ran" + (tests.executed > 0 ? " (" + tests.passing + " passed):" : "."));
+            testList.appendChild(errorTestsSummary);
+          } else {
+            summary.textContent = tests.executed == 1 && tests.passing == 1 ? "The test in this block passed."
```

Comment:
Please don't use quadruply-nested ternary operators here.  This is almost illegible in JS, even if it's pretty much standard Pyret style to pull the assignment/binding out to the topmost expression ;-)  Use a 4-line if/else if instead.